### ISSUE TITLE
RAM Usage

### ISSUE THREAD

**Issue by burdi01** [brndnmtthws/conky#920]
Slackware Current64 - Kernel 5.4.6 - Conky 1.11.6_pre Git 26d23cb Nov 1, 2019

With Conky 1.11.5 I had the problem as described in the closed issue #909. So I downloaded and built the current git version. Indeed the #909 problem is gone.

However in all those years that I run Conky I have never been able to correlate the RAM Usage / Available values to the ones given by running  "free". E.g. now on a 4 GB box I see 1.57 GiB (40%) / 3.85 GiB while "free" shows:
```
rescue:~# free -h 
            total        used        free      shared  buff/cache   available
Mem:        3.8Gi       348Mi       1.8Gi       855Mi       1.7Gi       2.3Gi
Swap:          0B          0B          0B
rescue:~#
```
Any hints would be appreciated.
:D

```lua
conky.config = {
-- the list of variables has been removed from this file in favour
-- of keeping the documentation more maintainable.
-- Check http://conky.sf.net for an up-to-date-list.

-- set to yes if you want Conky to be forked in the background
	background = true,

-- X font when Xft is disabled, you can pick one with program xfontsel
--font 5x7
--font 6x10
	font = '7x10',
--font 8x13
--font 9x15
--font *mintsmild.se*
--font -*-*-*-*-*-*-34-*-*-*-*-*-*-*

-- Use Xft?
	use_xft = false,

-- Xft font when Xft is enabled
	font = 'Bitstream Vera Sans Mono:size=11',

-- Text alpha when using Xft
	xftalpha = 0.8,

-- Print everything to stdout?
-- out_to_console no

-- Print everything to console?
-- out_to_console no

-- Update interval in seconds
	update_interval = 5.0,

-- This is the number of times Conky will update before quitting.
-- Set to zero to run forever.
	total_run_times = 0,

--own_window_title Parted Magic - conky

-- Create own window instead of using desktop (required in nautilus)
	own_window = true,

-- If own_window is yes, you may use type normal, desktop or override
	own_window_type = 'normal',

-- Use pseudo transparency with own_window?
	own_window_transparent = true,

-- If own_window_transparent is set to no, you can set the background colour here
	own_window_colour = 'black',

-- If own_window is yes, these window manager hints may be used
	own_window_hints = 'undecorated,below,sticky,skip_taskbar,skip_pager',
--own_window_hints below,skip_taskbar,skip_pager

-- Use double buffering (reduces flicker, may not work for everyone)
	double_buffer = true,

-- Minimum size of text area
	minimum_width = 220, minimum_height = 5,

-- Draw shades?
	draw_shades = true,

-- Draw outlines?
	draw_outline = false,

-- Draw borders around text
	draw_borders = false,

-- Draw borders around graphs
	draw_graph_borders = true,

-- Stippled borders?
	stippled_borders = 4,

-- border margins
	border_inner_margin = 4,

-- border width
	border_width = 1,

-- Default colors and also border colors
	default_color = 'white',
	default_shade_color = 'black',
	default_outline_color = 'black',

-- Text alignment, other possible values are commented
--alignment top_left
	alignment = 'top_right',
--alignment bottom_left
--alignment bottom_right
--alignment none

-- Gap between borders of screen and text
-- same thing as passing -x at command line
	gap_x = 15,
	gap_y = 15,

-- Subtract file system buffers from used memory?
	no_buffers = true,

-- set to yes if you want all text to be in uppercase
	uppercase = false,

-- number of cpu samples to average
-- set to 1 to disable averaging
	cpu_avg_samples = 2,

-- number of net samples to average
-- set to 1 to disable averaging
	net_avg_samples = 2,

-- Force UTF8? note that UTF8 support required XFT
	override_utf8_locale = false,

-- Add spaces to keep things from moving about?  This only affects certain objects.
	use_spacer = 'none',

};

conky.text = [[
${color goldenrod}Hostname: $nodename
${color goldenrod}Linux Kernel: $kernel
${color goldenrod}CPU Details:  $machine, $freq(MHz)

${color}CPU History:  ${color darkgreen}${cpugraph 30,0 0000ff 00ff00}
${color}CPU Usage:${color magenta2} $cpu% ${cpubar 11,0}

${color}RAM Usage:${color green} $mem ($memperc%) ${membar 11,0}
${color}RAM Total:${color green} $memmax
${if_match "$battery" != ""} 
${color}ACPI Battery: ${color DeepSkyBlue}$battery_short ${battery_bar 11,0} 
${color}Remaining time: ${color DeepSkyBlue}$battery_time
$endif
$color$stippled_hr
$alignc${color}Processes:$color $processes  ${color}Running:$color $running_processes
$alignc${color}(top 5 sorted by CPU usage)
${color goldenrod} NAME              PID    CPU%   MEM%
${color} ${top name 1} ${top pid 1} ${top cpu 1} ${top mem 1}
${color} ${top name 2} ${top pid 2} ${top cpu 2} ${top mem 2}
${color} ${top name 3} ${top pid 3} ${top cpu 3} ${top mem 3}
${color} ${top name 4} ${top pid 4} ${top cpu 4} ${top mem 4}
${color} ${top name 5} ${top pid 5} ${top cpu 5} ${top mem 5}

$alignc${color}(top 5 sorted by MEM usage)
${color goldenrod} NAME              PID    CPU%   MEM%
${color} ${top_mem name 1} ${top_mem pid 1} ${top_mem cpu 1} ${top_mem mem 1}
${color} ${top_mem name 2} ${top_mem pid 2} ${top_mem cpu 2} ${top_mem mem 2}
${color} ${top_mem name 3} ${top_mem pid 3} ${top_mem cpu 3} ${top_mem mem 3}
${color} ${top_mem name 4} ${top_mem pid 4} ${top_mem cpu 4} ${top_mem mem 4}
${color} ${top_mem name 5} ${top_mem pid 5} ${top_mem cpu 5} ${top_mem mem 5}
$color$stippled_hr
$alignc${color}System Uptime:${color DarkOrange1} $uptime
]];
```

**Comment by lasers** [comment]
I dunno RAM. My lazy guess is this... Convert em all to MiB.
```
total: (3.8 GiB)      = 3891.2 MiB
used:                 = 348    MiB
free: (1.8 GiB)       = 1843.2 MiB
shared:               = 855    MiB
buff/cache: (1.7 GiB) = 1740.8 MiB
available: (2.3 GiB)  = 2355.2 MiB
```

* available / total = 0.605% available... so used would be 0.39473%.

Otherwise, you can look at source code https://github.com/brndnmtthws/conky/blob/master/src/linux.cc#L176-L264 Cheers.

**Comment by burdi01** [comment]
That answers my question. Using the free values (the free -h ones are too coarse):
   usage = (total - available)  / 1024 / 1024
Thank you.
:D
